src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -181,4 +276,23 @@ export class ResultComponent implements OnInit, OnChanges {
         }
         return ['far', 'times-circle'];
     }
+
+    /**
+     * Find latest result in results array
+     *
+     * @param {Result} results
+     */
+    getLatestResult(results: Result[]) {
+        results.sort((r1: Result, r2: Result) => {
+            if (r1.completionDate! > r2.completionDate!) {
+                return -1;
+            }
+            if (r1.completionDate! < r2.completionDate!) {
+                return 1;
+            }
+            return 0;
+        });
+
+        return results[0];
```

Comment:
Fails if results is empty.

Reply:
I removed that method as I'm now using `reduce` to sort the array. There's a previous `this.hasParticipationResults()` check.